### PR TITLE
fix: stop attempting to double discount

### PR DESCRIPTION
**Pull request by JFrankfurt**
This also fixes the basedoteth discount

### CODE REVIEW

**Comment by cb-heimdall** [comment]
<h1>🟡 Heimdall Review Status</h1>
<table>
<tr>
<th>Requirement</th>
<th>Status</th>
<th>More Info</th>
</tr>
<tr>
<td>Reviews</td>
<td>
🟡
<code>0/1</code>
</td>
<td>
<details>
<summary>Denominator calculation</summary>
<table>
<tr>
<td>
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><code>1</code> if user is bot</td>
<td>0</td>
</tr>
<tr>
<td><code>1</code> if user is external</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>.codeflow.yml</code>
</td>
<td>1</td>
</tr>
<tr>
<td>
Additional review requirements
<details>
<summary>Show calculation</summary>
<table>
<tr>
<td><strong>Max</strong></td>
<td><strong>0</strong></td>
</tr>
</table>
</details>
</td>
<td>0</td>
</tr>
<tr>
<td>
From <code>CODEOWNERS</code>
</td>
<td>0</td>
</tr>
<tr>
<td>Global minimum</td>
<td>0</td>
</tr>
<tr>
<td><strong>Max</strong></td>
<td>
<strong>
1
</strong>
</td>
</tr>
</table>
</details>
</td>
<td>1</td>
</tr>
<tr>
<td><code>1</code> if commit is unverified</td>
<td>0</td>
</tr>
<tr>
<td><strong>Sum</strong></td>
<td><strong>1</strong></td>
</tr>
</table>
</details>
</td>
</tr>
</table>

File: apps/web/src/hooks/useAttestations.ts [hunk 1]
```diff
@@ -353,39 +352,75 @@ export function useBuildathonAttestations() {
...
+    useState<MerkleTreeProofResponse | null>(null);
+  const { logError } = useErrors();
 
-  const discountValidatorAddress = BASE_DOT_ETH_ERC721_DISCOUNT_VALIDATOR[basenameChain.id];
+  useEffect(() => {
+    async function checkBaseDotEthAttestations(a: string) {
+      try {
+        setLoading(true);
+        const params = new URLSearchParams();
+        params.append('address', a);
```
Comment (zencephalon) [inline apps/web/src/hooks/useAttestations.ts:369]: Typo here?

File: apps/web/src/hooks/useAttestations.ts [hunk 2]
```diff
@@ -353,39 +352,75 @@ export function useBuildathonAttestations() {
...
         discount: Discount.BASE_DOT_ETH_NFT,
-        validationData: '',
+        validationData: encodedProof,
       },
       loading: false,
       error: null,
     };
   }
-  return { data: null, loading: isLoading, error };
+  return { data: null, loading: loading || isLoading, error };
```
Comment (zencephalon) [inline apps/web/src/hooks/useAttestations.ts:423]: nit, Would be nice to distinguish the two loadings by name a little

File: apps/web/src/hooks/useAttestations.ts (hunk 1 above)
Comment (zencephalon) [reply apps/web/src/hooks/useAttestations.ts:369]: ah, a is the parameter string, k

**Review by zencephalon — DISMISSED** [review]

**Review by zencephalon — APPROVED** [review]

File: apps/web/src/hooks/useAttestations.ts [hunk 3]
```diff
@@ -353,39 +352,75 @@ export function useBuildathonAttestations() {
...
+  );
 
   const readContractArgs = useMemo(() => {
     if (!address) {
       return {};
     }
     return {
-      address: discountValidatorAddress,
+      address: baseDotEthProofResponse?.address,
       abi: ERC721ValidatorABI,
```
Comment (ricardoMogg) [inline apps/web/src/hooks/useAttestations.ts:404]: We should be using the baseEth ABI. (probably can reuse the cbid one)

Comment (ricardoMogg) [inline apps/web/src/hooks/useAttestations.ts:404]:
```suggestion
      abi: CBIDValidatorABI,
```

**Review by ricardoMogg — APPROVED** [review]

**Pull request merged**